portality/tasks/harvester_helpers/workflow.py

```diff
@@ -1,42 +1,47 @@
+from portality.api.v2 import ArticlesCrudApi
 from portality.api.v2.client import client as doajclient
 from portality.core import app
 from portality.lib import plugin
 from portality.decorators import capture_sigterm
+from portality.models import Journal, Account
 from portality.models.harvester import HarvesterProgressReport as Report
 from portality.models.harvester import HarvestState
 from portality.lib.dataobj import DataObjException
 
 
 class HarvesterWorkflow(object):
 
+    job = None
+
     @classmethod
     @capture_sigterm
-    def process_account(cls, account_id):
-        app.logger.info("Harvesting for Account:{x}".format(x=account_id))
-
-        doaj = doajclient.DOAJv1API()
-        gen = doaj.field_search_iterator("journals", "username", account_id)
-        issns = []
-        for journal in gen:
-            issns += journal.all_issns()
-        issns = list(set(issns))
-
-        app.logger.info("Account:{x} has {y} issns to harvest for: {z}".format(x=account_id, y=len(issns), z=",".join(issns)))
+    def process_account(cls, account_id, job):
+        cls.job = job
+        cls.job.add_audit_message("Harvesting for Account:{x}".format(x=account_id))
+        issns = cls.get_journals_issns(account_id)
+        cls.job.add_audit_message("Account:{x} has {y} issns to harvest for: {z}".format(x=account_id, y=len(issns), z=",".join(issns)))
 
         # now update the issn states
         HarvesterWorkflow.process_issn_states(account_id, issns)
 
         for issn in issns:
             HarvesterWorkflow.process_issn(account_id, issn)
 
+    @classmethod
+    def get_journals_issns(cls, account_id):
+        return Journal.issns_by_owner(account_id)
```

Comment:
An observation from another thread of work - we should probably only consider journals that are `in_doaj`, so you can add `in_doaj=False` to this method call, I think.

Reply:
ok, but I will add that as the default parameter so it could be called differently if needed.

Reply:
The `Journal.issns_by_owner` method does not accept `in_doaj` param:
```
@classmethod
    def issns_by_owner(cls, owner):
        q = IssnQuery(owner)
        res = cls.query(q=q.query())
        issns = [term.get("term") for term in res.get("facets", {}).get("issns", {}).get("terms", [])]
        return issns
```
I think it is actually a good idea to have it. I'll add the implementation.